pandas/tests/test_algos.py

```diff
@@ -2282,6 +2282,11 @@ def test_index(self):
             # algos.mode expects Arraylike, does *not* unwrap TimedeltaIndex
             algos.mode(idx)
 
+    def test_ser_mode_with_name(self):
+        ser = Series([1, 2, 3])
+        ser.name = "foo"
+        tm.assert_equal(ser.mode().name, "foo")
```

Comment:
Instead here could you compare the whole `mode()` result to an expected Series result?

```
result = ser.mode()
expected = Series(..., name="foo")
tm.assert_series_equal(result, expected)
```